src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -428,9 +428,16 @@ class ArrayLiteralSegment(BaseSegment):
     """An array literal segment."""
 
     type = "array_literal_type"
```

Comment:
I realise this is separate to your PR, but shouldn't this just be `array_literal`?

Reply:
Yes, I think so. I'll change it unless something blocks me (unlikely).